spyderlib/widgets/tests/test_array_builder.py

```diff
@@ -0,0 +1,204 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright © 2009- The Spyder Development Team
+# Licensed under the terms of the MIT License
+#
+
+"""
+"""
+
+# Third party imports
+from qtpy.QtCore import Qt
+from pytestqt import qtbot  # analysis:ignore
+
+# Local imports
+from spyderlib.widgets.arraybuilder import NumpyArrayDialog
+
+
+class TestNumpyArrayBuilder:
+    def test_array_inline_array(self, qtbot):  # analysis:ignore
+        dlg = NumpyArrayDialog(inline=True)
+        dlg.show()
+        qtbot.addWidget(dlg)
+
+        qtbot.keyClicks(dlg.widget, '1 2 3  4 5 6')
+        qtbot.keyPress(dlg.widget, Qt.Key_Return)
+
+        expected_value = 'np.array([[1, 2, 3],\n          [4, 5, 6]])'
+        value = dlg.text()
+        assert expected_value == value
```

Comment:
I would put them in the reverse order, I usually check if what I have is what I want, not the other way around.